src/XrdCms/XrdCmsCluster.cc

```diff
@@ -1786,10 +1788,15 @@ XrdCmsNode *XrdCmsCluster::SelbyLoad(SMask_t mask, XrdCmsSelector &selR)
     XrdCmsNode *np, *sp = 0;
     bool Multi = false, reqSS = (selR.needSpace & XrdCmsNode::allowsSS) != 0;
 
-// Scan for a node (preset possible, suspended, overloaded, full, and dead)
-//
-   selR.Reset(); SelTcnt++;
-   for (int i = 0; i <= STHi; i++)
+    // Scan for a node (preset possible, suspended, overloaded, full, and dead)
+    //
+    selR.Reset(); SelTcnt++;
+    int selCap = 1;
+    int randomSel=1;
+    //default 0 to skip the node in random selection if the below checks fail
+    int *weighed = new int[STHi] {0};
```

Comment:
I'd change this to allow for better optimization to:
int weighed = (Config.P_randlb==1 ? new int[STHi] {0} : 0);

Then subsequently substitute if (Config.P_randlb==1) with if (weighted)

Reply:
Or even better, if this is called frequently, it should not contain any allocations. Allocations should happen only at initialization.

Reply:
thanks, included in the second commit